crates/mun_memory/src/type_info.rs

```diff
@@ -130,6 +148,22 @@ impl TypeInfo {
             data: PointerInfo { pointee, mutable }.into(),
             immutable_pointer_type: Default::default(),
             mutable_pointer_type: Default::default(),
+            array_type: Default::default(),
+        })
+    }
+
+    /// Constructs a new array type
+    pub fn new_array(element: Arc<TypeInfo>) -> Arc<TypeInfo> {
+        Arc::new(TypeInfo {
+            name: format!("[{}]", &element.name),
+            layout: Layout::new::<*const std::ffi::c_void>(),
```

Comment:
Is there a reason you use `*const c_void`?

Reply:
Actually, it doesn't really make sense to have a layout for an array. The layout has a double meaning:

There are two types of layouts:
1. The layout to allocate the object on the heap.
2. The layout of a reference to a value of type T.

Layout in this context means the first, we store how many bytes (and with which alignment) the data for a type needs on the GC heap. For GC structs that is different than (2) because to store a reference another layout is needed (a pointer layout).

For arrays (1) doesn't make any sense because it depends both on the layout of the elements as well as the capacity of the array. We should probably refactor this at some point to make a clear distinction between the two.

Reply:
I agree. For arrays we could just use a `const` layout, saving memory. Same applies for pointers (in the future I expect). Could you outline this in a new issue?

Reply:
See #443